velox/docs/develop/TpchBenchmark.rst

```diff
@@ -0,0 +1,203 @@
+=============================================================
+
+**num_drivers**
+---------------
+
+This configuration option describes the number of resources available to
+process query plan tasks. This can greatly improve performance of CPU bound
+workloads. The recommendation in the table above seems to be optimal for
+TPCH-like workloads.
+
+**num_io_threads**
+------------------
+
+This configuration option describes the number of threads available per driver
+to retrieve data from the network source. If the workload is I/O bound, then
+increasing this number is beneficial if the data is fewer requests of larger
+chunks as opposed to many smaller requests.
+
+**cache_gb**
+------------
+
+This configuration option is useful for workloads that read the same data
+several times per query but only applies to the single process use case.
+*NOTE: There is a SSD Caching option in Velox but it to is ONLY useful in
+the single process use case.*
+
+**num_splits_per_file**
+-----------------------
+
+This configuration option is best when the data set count of row groups
+matches this value. The affect in overall performance appears based on
+testing to be small, however.
+
+Optimizations for All Workloads (Both Use Cases)
+================================================
+
+**max_coalesce_bytes**
+----------------------
+
+This configuration option is the maximum bytes coalesced into a single request
+to the data source. This was tested from the default 128MB to 2GB, and the
+overall improvement was small as size increased. Capturing request data did
+show larger and fewer requests but not enough to vastly improve I/O performance.
+
+**max_coalesce_distance_bytes**
+-------------------------------
+
+This configuration option is the maximum byte distance between needed data in
+the same file at the data source that can be coalesced. Increasing this value
+would theoretically reduce the number of requests and increase each request
+size. However, if made too large the query will return too many un-needed bytes
+and could decrease I/O performance. This plus __max_coalesce_bytes__ should be
+fine-tuned for the workload being run.
+
+Summary
+=======
+
+If a use of Velox matches the use case of the TcphBenchmark then it is a good
+tool to test, I/O and driver performance for specific TCP-H queries. This would
+benefit execution of specific production workloads that are like the chosen
+queries. If in multi-process use case, like Spark/Gluten/Velox configuration,
+the recommendation is to oversubscribe I/O threads between 2X and 3X vCPUs and
+tune the 2 coalesce configurations exposed.
+
+----
+
+Appendix A: TpchBenchmark Tool Help Output
+==========================================
+
+From the repository root, use the following command line to see all the
+available flags in the TpchBenchmark tool.
+
+.. code:: shell
+
+   $ ./_build/release/velox/benchmarks/tpch/velox_tpch_benchmark --help
```

Comment:
nit: this should be `./_build/release/velox/benchmarks/tpch/velox_tpch_benchmark -helpon=TpchBenchmark` to see only the TpchBenchmark flags.

Reply:
@paul-amonson Can you ping me after fixing this nit and I will start the merge process.

Reply:
@kgpai Ok Done.